src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -0,0 +1,139 @@
+register_cli_argument('sf', 'cluster_name',options_list=('--name', '--cluster-name','-n'),
+                      help='Specify the name of the cluster, if not given it will be same as resource group name')
+register_cli_argument('sf', 'location',
+                      validator=get_default_location_from_resource_group)
+
+register_cli_argument('sf', 'secret_identifier',options_list=('--secret-identifier', '-si'),
+                      help='The existing Azure key vault secret URL')
+register_cli_argument('sf', 'certificate_file',options_list=('--certificate-file', '-cf'),
+                     help='The existing certificate file path for the primary cluster certificate.')
+register_cli_argument('sf', 'parameter_file',options_list=('--parameter-file', '-pf'),
+                     help='The path to the template parameter file.')
+register_cli_argument('sf', 'template_file',options_list=('--template-file', '-tf'),
+                     help='The path to the template file.')
+register_cli_argument('sf', 'vm_password',options_list=('--vm-password', '-vp'),
+                      help='The password of the Vm')
+register_cli_argument('sf', 'certificate_output_folder',options_list=('--certificate-output-folder', '-cof'),
+                      help='The folder of the new certificate file to be created.')
+register_cli_argument('sf', 'certificate_password',options_list=('--certificate-password', '-cp'),
+                      help='The password of the certificate file.')
+register_cli_argument('sf', 'certificate_subject_name',options_list=('--certificate-subject-name', '-sn','-csn'),
+                      help='The subject name of the certificate to be created.')
+register_cli_argument('sf', 'vault_resource_group_name',options_list=('--vault-resource-group', '-vg','-vrg'),
+                      help='Azure key vault resource group name, it not given it will be defaulted to cluster resource group name')
+register_cli_argument('sf', 'vault_name',options_list=('--vault-name', '-vn'),
+                      help='Azure key vault name, it not given it will be defaulted to the cluster resource group name')
+register_cli_argument('sf', 'cluster_size',options_list=('--cluster-size', '-size','-s'),
+                      help='The number of nodes in the cluster. Default are 5 nodes')
+register_cli_argument('sf', 'vm_sku', options_list=('--vm-sku', '-sku'), help='The Vm Sku')
+register_cli_argument('sf', 'vm_user_name', options_list=('--vm-user-name', '-un'), help='The user name for logging to Vm. Default will be adminuser')
+register_cli_argument('sf', 'vm_os', default='WindowsServer2016Datacenter', options_list=('--vm-os', '-os'),
+                      help='The Operating System of the VMs that make up the cluster.',
+                     **enum_choice_list(['WindowsServer2012R2Datacenter', 'WindowsServer2016Datacenter', 'WindowsServer2016DatacenterwithContainers', 'UbuntuServer1604']))
```

Comment:
This should use a strong enum imported from the SDK instead of manually listing the strings (unless there is no such enum).